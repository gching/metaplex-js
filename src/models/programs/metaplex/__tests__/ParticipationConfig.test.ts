import {
  NonWinningConstraint,
  ParticipationConfig,
  WinningConstraint,
} from '../ParticipationConfig';

describe('ParticipationConfig', () => {
  it('should instantiate with the correct values', () => {
    const inst = new ParticipationConfig({
      fixedPrice: undefined,
      nonWinningConstraint: NonWinningConstraint.GivenForFixedPrice,
      safetyDepositBoxIndex: 5,
      winnerConstraint: WinningConstraint.NoParticipationPrize,
    });

    expect(inst).toMatchSnapshot();
  });
});
