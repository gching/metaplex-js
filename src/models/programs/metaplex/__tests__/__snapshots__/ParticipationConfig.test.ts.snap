// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipationConfig should instantiate with the correct values 1`] = `
ParticipationConfig {
  "fixedPrice": undefined,
  "nonWinningConstraint": 1,
  "safetyDepositBoxIndex": 5,
  "winnerConstraint": 0,
}
`;
